src/librustc/mir/mod.rs

```diff
@@ -1131,23 +1134,7 @@ impl<'tcx> TerminatorKind<'tcx> {
                     write!(fmt, "!")?;
                 }
                 write!(fmt, "{:?}, ", cond)?;
-
-                match *msg {
-                    AssertMessage::BoundsCheck { ref len, ref index } => {
-                        write!(fmt, "{:?}, {:?}, {:?}",
-                               "index out of bounds: the len is {} but the index is {}",
-                               len, index)?;
-                    }
-                    AssertMessage::Math(ref err) => {
-                        write!(fmt, "{:?}", err.description())?;
-                    }
-                    AssertMessage::GeneratorResumedAfterReturn => {
-                        write!(fmt, "{:?}", "generator resumed after completion")?;
-                    }
-                    AssertMessage::GeneratorResumedAfterPanic => {
-                        write!(fmt, "{:?}", "generator resumed after panicking")?;
-                    }
-                }
+                write!(fmt, "\"{:?}\"", msg)?;
```

Comment:
You can combine these `write!` calls now.

Reply:
Also, isn't there a `description` method? Or `fmt::Display` impl? `Debug` seems the wrong thing to put the user-facing message into.

Reply:
done & we need debug, because we're recursively printing the `Operand`s, and they were printed with debug printing before.